src/unix/apple/system.rs

```diff
@@ -348,62 +349,56 @@ impl SystemInner {
         boot_time()
     }
 
-    pub(crate) fn name() -> Option<String> {
+    pub(crate) fn name() -> Option<OsString> {
         get_system_info(libc::KERN_OSTYPE, Some("Darwin"))
     }
 
-    pub(crate) fn long_os_version() -> Option<String> {
+    pub(crate) fn long_os_version() -> Option<OsString> {
         #[cfg(target_os = "macos")]
-        let friendly_name = match Self::os_version().unwrap_or_default() {
-            f_n if f_n.starts_with("14.0") => "Sonoma",
-            f_n if f_n.starts_with("10.16")
-                | f_n.starts_with("11.0")
-                | f_n.starts_with("11.1")
-                | f_n.starts_with("11.2") =>
+        let friendly_name = match Self::os_version().unwrap_or_default().as_bytes() {
+            f_n if f_n.starts_with(b"14.0") => "Sonoma",
+            f_n if f_n.starts_with(b"10.16")
+                | f_n.starts_with(b"11.0")
+                | f_n.starts_with(b"11.1")
+                | f_n.starts_with(b"11.2") =>
             {
                 "Big Sur"
             }
-            f_n if f_n.starts_with("10.15") => "Catalina",
-            f_n if f_n.starts_with("10.14") => "Mojave",
-            f_n if f_n.starts_with("10.13") => "High Sierra",
-            f_n if f_n.starts_with("10.12") => "Sierra",
-            f_n if f_n.starts_with("10.11") => "El Capitan",
-            f_n if f_n.starts_with("10.10") => "Yosemite",
-            f_n if f_n.starts_with("10.9") => "Mavericks",
-            f_n if f_n.starts_with("10.8") => "Mountain Lion",
-            f_n if f_n.starts_with("10.7") => "Lion",
-            f_n if f_n.starts_with("10.6") => "Snow Leopard",
-            f_n if f_n.starts_with("10.5") => "Leopard",
-            f_n if f_n.starts_with("10.4") => "Tiger",
-            f_n if f_n.starts_with("10.3") => "Panther",
-            f_n if f_n.starts_with("10.2") => "Jaguar",
-            f_n if f_n.starts_with("10.1") => "Puma",
-            f_n if f_n.starts_with("10.0") => "Cheetah",
+            f_n if f_n.starts_with(b"10.15") => "Catalina",
+            f_n if f_n.starts_with(b"10.14") => "Mojave",
+            f_n if f_n.starts_with(b"10.13") => "High Sierra",
+            f_n if f_n.starts_with(b"10.12") => "Sierra",
```

Comment:
I remove older versions, because they are unsupported by Rust now. Also I renamed it to `macOS` in the string, as that's the official way to write the name now.

Reply:
Please leave them as other rust backends still support them.

Reply:
Well, this PR as a whole (I read that you want it to be split up, so I'll take that into account) would require 1.74, which does not support any older versions of macOS.